Core/SubterfugeCore/Core/Players/PlayerCurrency.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using SubterfugeCore.Core.Players;
+
+namespace SubterfugeCore.Core.Players
+{
+	public class PlayerCurrency
+	{
+		/// <summary>
+		/// A table of all the player's currencies
+		/// </summary>
+		private Dictionary<String, Dictionary<String,Object>> Currencies = new Dictionary<String, Dictionary<String, Object>>();
+
+		/// <summary>
+		/// Creates a new currency as long as one with the same name doesn't already exist. Returns false if failed.
+		/// </summary>
+		/// <param name="currencyName">The name of the currency</param>
+		/// <param name="currencyValue">The inital value of the currency</param>
+		/// <param name="canBeNegative">Can the currency go into the negatives</param>
+		/// <returns>True if succeeded</returns>
+		public bool CreateCurrency(String currencyName, int currencyValue, bool canBeNegative)
```

Comment:
Instead of needing to create the currency, we can just create it on use. So keep this function around, but make it private and call it on inserts and updates if it doesn't already exist.

Reply:
👍